extension/js/common/oauth2/oauth2_inject.ts

```diff
@@ -0,0 +1,10 @@
+/* ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com */
+
+// Decleared win variable to avoid `Type 'string' is not assignable to type 'Location | (string & Location)'.ts(2322)` error
+const win: Window = window;
+
+// Redirect back to the extension itself so that we have priveledged
```

Comment:
Have you tried to add `frame-ancestors https://www.google.com/robots.txt 'self'` (or the like) to manifest's `"content_security_policy"`. I guess it'll let you oauth2_finish right away.

We have this line `frame-ancestors https://mail.google.com 'self'` already and it is also overriden for mocks with https://github.com/FlowCrypt/flowcrypt-browser/blob/59bc0b3890294f05bd41f1bcf0eef6dbd25294ca/tooling/build-types-and-manifests.ts#L128

Reply:
Sorry but not sure exactly what you mean.
AFAIK, frame-ancestors directive specifies valid parents that may embed a page(using iframe or something like that).
I'm not sure how it would let oauth2_finish right away.

Reply:
What was the error preventing finishing OAuth2 directly from `robots.txt`?
There is a number of security options, maybe enabling any of them can help.

Reply:
What I am trying to understand: what prevents us sending a message to close the oauth2 dialog directly from `robots.txt`?

Reply:
Like here:
```
// Even accepts arrays of patterns!
chrome.tabs.query({url: "*://*.example.com/*"}, function(tabs) {
  tabs.forEach(function(tab) {
    chrome.tabs.sendMessage(tab.id, message);
  });
});
```
https://stackoverflow.com/a/28832962

Reply:
`robots.txt` page url contains redirect_url which contains `authorization_code`, `scopes`, etc.
Our final goal is to get this url when we call `GoogleAuth.oauthLogin`.
But without help of content script injection (`oauth2_inject.ts` for our case), we can't get that url.

Reply:
Yes, we get this data in `oauth2_inject.ts`. Can we send it to the tab that awaits it (has set up the listener for it?)

Reply:
Can `robots.txt` page access `chrome.tabs` or `chrome.extension.getViews`?
I don't think it would.
> Like here:

Reply:
![image](https://user-images.githubusercontent.com/77344191/188907239-68084770-07c4-4887-8931-1e233255ac5b.png)